big_tests/tests/sm_SUITE.erl

```diff
@@ -346,19 +346,19 @@ resend_unacked_on_reconnection(Config) ->
     Messages = [<<"msg-1">>, <<"msg-2">>, <<"msg-3">>],
     {Bob, _} = given_fresh_user(Config, bob),
     {Alice, AliceSpec0} = given_fresh_user(Config, alice),
-        discard_vcard_update(Alice),
-        %% Bob sends some messages to Alice.
-        [escalus:send(Bob, escalus_stanza:chat_to(Alice, Msg))
-         || Msg <- Messages],
-        %% Alice receives the messages.
-        Stanzas = escalus:wait_for_stanzas(Alice, length(Messages)),
-        [escalus:assert(is_chat_message, [Msg], Stanza)
-         || {Msg, Stanza} <- lists:zip(Messages, Stanzas)],
-        %% Alice disconnects without acking the messages.
+    discard_vcard_update(Alice),
+    %% Bob sends some messages to Alice.
+    [escalus:send(Bob, escalus_stanza:chat_to(Alice, Msg))
+     || Msg <- Messages],
+    %% Alice receives the messages.
+    Stanzas = escalus:wait_for_stanzas(Alice, length(Messages)),
+    [escalus:assert(is_chat_message, [Msg], Stanza)
+     || {Msg, Stanza} <- lists:zip(Messages, Stanzas)],
+    %% Alice disconnects without acking the messages.
+    {ok, C2SRef} = monitor_terminating_process(AliceSpec0),
     escalus_connection:stop(Alice),
+    ok = wait_for_process_termination(C2SRef),
```

Comment:
Please add a comment stating that there's a race condition between the C2S process and `mod_offline` which will redirect to this PR for details.